src/standalone/userJupyterServer/userServerUrlProvider.ts

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+'use strict';
+import { inject, injectable } from 'inversify';
+import { Disposable, QuickInputButtons, QuickPickItem, Uri, window } from 'vscode';
+import { JupyterConnection } from '../../kernels/jupyter/jupyterConnection';
+import { validateSelectJupyterURI } from '../../kernels/jupyter/serverSelector';
+import { IJupyterServerUri, IJupyterUriProvider, IJupyterUriProviderRegistration } from '../../kernels/jupyter/types';
+import { IExtensionSyncActivationService } from '../../platform/activation/types';
+import { IApplicationShell, IClipboard, IEncryptedStorage } from '../../platform/common/application/types';
+import { Settings } from '../../platform/common/constants';
+import { IConfigurationService, IDisposable, IsWebExtension } from '../../platform/common/types';
+import { DataScience } from '../../platform/common/utils/localize';
+
+export const UserJupyterServerUriListKey = 'user-jupyter-server-uri-list';
+
+@injectable()
+export class UserJupyterServerUrlProvider implements IExtensionSyncActivationService, IDisposable, IJupyterUriProvider {
+    readonly id: string = '_builtin.jupyterServerUrlProvider';
+    readonly displayName: string = DataScience.UserJupyterServerUrlProviderDisplayName();
+    readonly detail: string = DataScience.UserJupyterServerUrlProviderDetail();
+    private _servers: { handle: string; serverInfo: IJupyterServerUri }[] = [];
+    private _initialized: Promise<void> | undefined;
+
+    constructor(
+        @inject(IClipboard) private readonly clipboard: IClipboard,
+        @inject(IJupyterUriProviderRegistration)
+        private readonly uriProviderRegistration: IJupyterUriProviderRegistration,
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IConfigurationService) private readonly configService: IConfigurationService,
+        @inject(JupyterConnection) private readonly jupyterConnection: JupyterConnection,
+        @inject(IsWebExtension) private readonly isWebExtension: boolean,
+        @inject(IEncryptedStorage) private readonly encryptedStorage: IEncryptedStorage
+    ) {}
+
+    activate() {
+        this.uriProviderRegistration.registerProvider(this);
+        this._initialized = new Promise(async (resolve) => {
+            const cache = await this.encryptedStorage.retrieve(
```

Comment:
We probably need to do a migration: copy servers manually provided from `serverUriStorage` to this tiny extension but let's play around with it first and handle the migration a bit later.

Reply:
Isn't this a little risky,
Running this in `activate` will drastically slow extension loading times.
Accessing users credential store could be a very slow operation